pool/src/pool.rs

```diff
@@ -413,12 +413,12 @@ where
 						weighting,
 						self.verifier_cache.clone(),
 					) {
-						if new_bucket.fee_to_weight >= bucket.fee_to_weight {
-							// Only aggregate if it would not reduce the fee_to_weight ratio.
+						if new_bucket.fee_rate >= bucket.fee_rate {
+							// Only aggregate if it would not reduce the fee_rate ratio.
 							tx_buckets[pos] = new_bucket;
```

Comment:
If we now support tx aggregation with combination of overpaying fees and suitable `fee_shift` values - does this allow us to relax this aggregation rule?

From the RFC -

> For instance, suppose two transactions have the same minfee. If one pays 3 times the minfee with a fee shift of 1, while the other pays exactly minfee with a zero fee shift, then both can be aggregated into one that pays twice the joint minfee (which is double the old one).

Currently this aggregation rule allows later txs to aggregate (during bucketing) with earlier txs iff the `fee_rate` does not decrease.

Txs `A` and `B` here being added to the _same_ bucket because `B` spends an output from `A`.

Reply:
Yes, we should relax the aggregation rules to allow any aggregation that preserves shifted_fee compliance.
But I don't want to make this logic height dependent, so we can make that (non-consensus) change later.

Reply:
Yes. That makes sense. :+1: